linked_list/linked_list.py

```diff
@@ -1,100 +1,202 @@
 
 # Defines a node in the singly linked list
 class Node:
-
-    def __init__(self, value, next_node = None):
+    def __init__(self, value, next=None):
         self.value = value
-        self.next = next_node
+        self.next = next
 
 # Defines the singly linked list
+
+
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
+        self.head = None  # keep the head private. Not accessible outside this class
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
-    def get_first(self):
-        pass
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
 
+    def get_first(self):
+        if self.head is None:
+            return None
+        return self.head.value
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(self, value):
-        pass
+        new_node = Node(value)
+        if self.head is None:
+            self.head = new_node
+        else:
+            new_node.next = self.head
+            self.head = new_node
+        return self.head.value
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
+
     def search(self, value):
-        pass
+        if self.head is None:
+            return False
+        current_node = self.head
+        while current_node is not None:
+            if current_node.value == value:
+                return True
+            current_node = current_node.next
+        return False

```

Comment:
This could be *slightly* more compact

```suggestion
        current_node = self.head
        while current_node is not None:
            if current_node.value == value:
                return True
            current_node = current_node.next
        return False

```